tests/test_tornado.py

```diff
@@ -1,28 +1,34 @@
 import unittest
-from tornado import gen
+import tornado
+from tornado.testing import gen_test
 from tornado.ioloop import IOLoop
 from gremlinclient import submit
 
 
-class Py27SyntaxTest(unittest.TestCase):
+class Py27SyntaxTest(tornado.testing.AsyncTestCase):
 
     def setUp(self):
+        super(Py27SyntaxTest, self).setUp()
         self.loop = IOLoop.current()
 
+    @gen_test
     def test_submit(self):
 
-        @gen.coroutine
-        def go():
-            f = submit("1 + 1")
-            res = yield f
-            while True:
-                msg = yield res.read()
-                if msg is None:
-                    break
-                self.assertEqual(msg.status_code, 200)
-                self.assertEqual(msg.data[0], 2)
-
-        self.loop.run_sync(go)
+        f = submit("1 + 1")
+        res = yield f
+        msg = yield res.read()
+        self.assertEqual(msg.status_code, 200)
+        self.assertEqual(msg.data[0], 2)
+
+    @gen_test(timeout=1)
+    def test_exception(self):
+
+        with self.assertRaises(RuntimeError):
+            func = submit("throw new Exception('error')")
+            res = yield func
```

Comment:
maybe we could use ``f`` or ``fut`` here instead of ``func`` because what it returns is a future.